README.md

```diff
@@ -9,15 +9,285 @@ Turning API Responses into Views with Handlebars
+
+We need a way to have a pre-fabricated bay of shelves ready to go and
+receive a new stack of books. We need a *template*.
+
+## Handlebars Templatess
+
+[Handlebars](http://handlebarsjs.com/) is a template engine gives us a way to build HTML templates separately from
+our code and allow us to use those templates to dynamically update the DOM.
+
+Creating a Handlebars template is simple. We just create the HTML that
+we want within a `script` tag, and use the `{{}}` delimiters to hold
+places for data. Let's convert our repository template to Handlebars
+inside `index.html`.
+
+```html
+<script id="repository-template" type="text/x-handlebars-template">
+  <ul>
+   {{#each this}}
+     <li>
+       <h2><a href="{{html_url}}">{{name}}</a></h2>
+       <p>Watchers: {{watchers_count}}</p>
+       <p>Forks: {{forks_count}}</p>
+       <p>Issues: {{open_issues_count}}</p>
+     </li>
+    {{/each}}
+  </ul>
+</script>
+```
+
+Keep in mind that the variable names between the double-curly-brace
+delimiters should match the names of the properties of the context
+object, in this case, our JSON response.
+
+Note that rather than build each list item sepately in a loop, we're
+making use of the `{{#each}}` helper to build our list for us based on
+the array of objects we pass in.
+
+Now let's update our `showRepositories` code to use the template.
+
+```js
+function showRepositories(event, data) {
+  const repos = JSON.parse(this.responseText)
+  const src = document.getElementById("repository-template").innerHTML
+  const template = Handlebars.compile(src)
+  const repoList = template(repos)
+  document.getElementById("repositories").innerHTML = repoList
+}
+```
+
+We get the `innerHTML` of our template `script` tag to compile into a
+`template` function that we pass our JSON response into.
+
+Since we are giving the template function the entire collection of
+repository objects, adding new fields to our template is as easy as
+creating the markup.
+
+Let's add the owner information to our template.
+
+```html
+<script id="repository-template" type="text/x-handlebars-template">
+  <ul>
+   {{#each this}}
+     <li>
+       <h2><a href="{{html_url}}">{{name}}</a></h2>
+       <section>
+          <header><h4>Created By {{owner.login}}</h4></header>
+          <img src="{{owner.avatar_url}}" height="32" width="32">
+       </section>
+       <p>Watchers: {{watchers_count}}</p>
+       <p>Forks: {{forks_count}}</p>
+       <p>Issues: {{open_issues_count}}</p>
+     </li>
+    {{/each}}
+  </ul>
+</script>
+```
+
+If we reload our page, everything works! Handlebars allows us to access
+the entire object graph, so `{{owner.login}}` works just fine to access
+the `login` property of the `owner` property of the repository and we
+didn't have to make any JavaScript edits.
+
+And that's the power of using a template engine like Handlebars. The
+presentation, or HTML code, can be managed separately from the logic and
+data.
```

Comment:
I really like how decoupled this section is from the previous Handlebars lessons. It lets us reuse this lesson out of order if we can, for example, assume that students will grok the Handlebars bits easily (or in a shorter course).